nestkernel/connection.h

```diff
@@ -150,6 +150,18 @@ class Connection
   void set_status( const DictionaryDatum& d, ConnectorModel& cm );
 
   /**
+   * Check syn_spec dictionary for parameters that are not allowed with the
+   * given connection.
+   *
+   * Will issue warning or throw error if an illegal parameter is found. The
+   * method does nothing if no illegal parameter is found.
+   *
+   * @note Classes requiring checks need to override the function with their own
+   * implementation, as this base class implementation does not do anything.
+   */
+  void check_synapse_params( const DictionaryDatum& d ) const;
```

Comment:
@jakobj As far as I understand, the connection class is not supposed to have any virtual functions, because we then store pointers to the subclasses so we can use the correct overridden function.  By making this into an empty function, we avoid this memory allocation.

@heplesser does this sound about right?